### ISSUE TITLE
train

### ISSUE THREAD

**Issue by JC0624** [xuhongzuo/couta#3]
Hi, I would like to ask how to retrain this model. I tried to modify the code, but no matter how I modified it, the results were always the same as in the paper and did not change.

**Comment by xuhongzuo** [comment]
Hi, thanks for your interest.

We reported the averaged results of five independent runs, and we set 42 as the beginning random seed for better reproduction of our results in the paper. (see `main.py`)

- If you just want to see how the model fluctuates with different random settings, just modify `42` to a new number that you like.
- If you want to modify the model, you may need to change `couta_algo.py`.

**Comment by JC0624** [comment]
> Hi, thanks for your interest.
> 
> We reported the averaged results of five independent runs, and we set 42 as the beginning random seed for better reproduction of our results in the paper. (see `main.py`)
> 
> * If you just want to see how the model fluctuates with different random settings, just modify  to a new number that you like.`42`
> * If you want to modify the model, you may need to change .`couta_algo.py`

Thank you very much for your reply! Your work is very interesting and very helpful to my research area. I've tried modifying your model as well as the loss function, but what confuses me is that the results I get are always the ones in the paper.

**Comment by xuhongzuo** [comment]
> > Hi, thanks for your interest.
> > We reported the averaged results of five independent runs, and we set 42 as the beginning random seed for better reproduction of our results in the paper. (see `main.py`)
> > 
> > * If you just want to see how the model fluctuates with different random settings, just modify  to a new number that you like.`42`
> > * If you want to modify the model, you may need to change .`couta_algo.py`
> 
> Thank you very much for your reply! Your work is very interesting and very helpful to my research area. I've tried modifying your model as well as the loss function, but what confuses me is that the results I get are always the ones in the paper.

If you have modified the class in `couta_algo.py`, I think the results should change. Could you please double-check your experiment process? I am happy to help by reviewing a pull request if there is still something wrong.

**Comment by JC0624** [comment]
> > > 嗨，感谢您的关注。我们报告了五次独立运行的平均结果，并将42设置为起始随机种子，以便在论文中更好地再现我们的结果。（请参见`main.py`)
> > > 
> > > * 如果您只想查看模型如何随不同的随机设置波动，只需修改为您喜欢的新数字即可。`42`
> > > * 如果要修改模型，可能需要更改 。`couta_algo.py`
> > 
> > 
> > 非常感谢您的回复！你的工作非常有趣，对我的研究领域非常有帮助。我尝试过修改你的模型以及损失函数，但让我感到困惑的是，我得到的结果总是论文中的结果。
> 
> 如果你在 中修改了类，我认为结果应该会改变。您能仔细检查一下您的实验过程吗？如果仍然有问题，我很乐意通过审查拉取请求来提供帮助。`couta_algo.py`

1. You can see that this is the loss function from your paper and its corresponding result.
<img width="301" alt="1662093528117" src="https://user-images.githubusercontent.com/64187594/188060301-53363f09-c2f3-4b72-95b3-0a7092c6fceb.png">
<img width="283" alt="1662093528119" src="https://user-images.githubusercontent.com/64187594/188060310-842dffd9-5a79-49f5-bcd8-c0f58ae74f88.png">
2. But when I added the modules in NeuTraL AD and their corresponding loss functions, the results were the same as before.I don't know if there is a problem with the modified code, so I tried to remove the loss functions corresponding to the umc and nac modules, but the result is still unchanged.
<img width="381" alt="Snipaste_2022-09-02_12-54-23" src="https://user-images.githubusercontent.com/64187594/188061779-f977fb3f-9c3e-4b7d-914d-339442dab2f4.png">
<img width="273" alt="Snipaste_2022-09-02_12-54-35" src="https://user-images.githubusercontent.com/64187594/188061796-247f4012-328f-4fb8-bc60-f489f022dd2c.png">
<img width="128" alt="Snipaste_2022-09-02_12-56-40" src="https://user-images.githubusercontent.com/64187594/188061978-0fd9d57a-91fd-4fb3-a706-350e3ea4962b.png">
<img width="279" alt="Snipaste_2022-09-02_12-56-50" src="https://user-images.githubusercontent.com/64187594/188061983-9e21a2d5-3646-42c6-aaff-de03fb50a90f.png">

**Comment by xuhongzuo** [comment]
I saw the f1 score changed from 0.8227 to 0.8264 when adding `loss_usl`? 
Do you mean the result is not changed too much?

**Comment by JC0624** [comment]
> I saw the f1 score changed from 0.8227 to 0.8264 when adding `loss_usl`? Do you mean the result is not changed too much?

I'm sorry I didn't notice this. Since I encountered this problem yesterday, I modified the code again and may have solved the problem. I apologize for wasting your time and thank you again for your reply!

**Comment by xuhongzuo** [comment]
> > I saw the f1 score changed from 0.8227 to 0.8264 when adding `loss_usl`? Do you mean the result is not changed too much?
> 
> I'm sorry I didn't notice this. Since I encountered this problem yesterday, I modified the code again and may have solved the problem. I apologize for wasting your time and thank you again for your reply!

No worries! I wish you every success with your research. I closed this issue and please feel free to contact me if you have any further problems or questions. 
Hope you have a wonderful weekend! 

Cheers
